### PR TITLE
Yk sir changes

### PR DESCRIPTION
**Pull request by vext01**
Companion PR for https://github.com/softdevteam/yk/pull/29

### CODE REVIEW

**Comment by vext01** [comment]
The last commit breaks the CI cycle.

If you are happy, this is ready for the sceptical eye of bors.

**Comment by vext01** [comment]
Oops, sorry. Last push fixes. OK to squash?

**Comment by ptersilie** [comment]
Yes, please squash.

**Comment by vext01** [comment]
splat

**Comment by vext01** [comment]
```
hread 'rustc' panicked at 'assertion failed: num_mir_blks % 2 == 0', src/librustc_yk_sections/emit_sir.rs:136:17
```

OK, so I broke some of my own tests. Fixing.

**Comment by vext01** [comment]
That last push fixes this. If you are OK with it, I will squash.

File: src/librustc_yk_sections/emit_sir.rs [hunk 1]
```diff
@@ -130,13 +130,14 @@ impl<'a, 'tcx> ConvCx<'a, 'tcx> {
...
         // terminators due to this transformation. The user blocks will be pushed at new indices
         // which are already correct for the existing successor indices.
-        let skip = match self.tcx.sess.opts.cg.tracer {
-            TracerMode::Software => {
-                let num_mir_blks = self.mir.basic_blocks().len();
-                debug_assert!(num_mir_blks % 2 == 0);
-                num_mir_blks / 2
-            },
-            _ => 0,
+        let skip = if self.tcx.sess.opts.cg.tracer == TracerMode::Software &&
```
Comment (ltratt) [inline src/librustc_yk_sections/emit_sir.rs]: This is a bit hard to read. Would it be better as:

```rust
let skip;
if ... {
   skip = ...;
}
else {
   skip = ...;
}
```

?

File: src/librustc_yk_sections/emit_sir.rs [hunk 2]
```diff
@@ -478,3 +479,28 @@ fn lower_def_id(tcx: TyCtxt<'_>, &def_id: &DefId) -> ykpack::DefId {
...
+        }, ..
+    }, ..), ..} = first_block.terminator.as_ref().unwrap().kind {
+        // Block is terminated by a static call. So is it the trace recorder?
+        let rec_fn_defid = tcx.get_lang_items(LOCAL_CRATE).yk_swt_rec_loc()
+            .expect("couldn't find software trace recorder function");
+        return *def_id == rec_fn_defid;
+    }
+
+    // Block wasn't terminated by a static call.
+    return false;
```
Comment (ltratt) [inline src/librustc_yk_sections/emit_sir.rs]: `return` isn't necessary here.

File: src/librustc_yk_sections/emit_sir.rs (hunk 1 above)
Comment (vext01) [reply src/librustc_yk_sections/emit_sir.rs]: I'm indifferent, so I'll update it :)

File: src/librustc_yk_sections/emit_sir.rs (hunk 2 above)
Comment (vext01) [reply src/librustc_yk_sections/emit_sir.rs]: true.

(well, no false, but true that I don't need the `return` 😆 )

**Comment by vext01** [comment]
Try that last push.

**Comment by ltratt** [comment]
Please squash.

**Comment by vext01** [comment]
splat.

**Comment by ptersilie** [comment]
@ltratt Since you have taken over, do you want to trigger bors or shall I?

**Comment by ltratt** [comment]
It doesn't matter :) But since I'm here...

bors r+

**Comment by vext01** [comment]
That's unfortunate. Restarting buildbot cancels running builds.

If you try again, this should work.

**Pull request merged**